layersvt/api_dump.h

```diff
@@ -833,389 +851,359 @@ class ApiDumpInstance {
-        settings.stream() << " (" << shaderDumpFileName.str() << ")\n";
-        ++shaderDumpIndex;
-        std::ofstream shaderDumpFile;
-        shaderDumpFile.open(shaderDumpFileName.str(), std::ofstream::out | std::ostream::trunc);
-        shaderDumpFile << stream.str() << "\n";
-        shaderDumpFile.close();
+        dump_text_value(array[i], settings, child_type, indexName.c_str(), indents + 1, dump);
     }
 }
 
-template <typename T, typename... Args>
-inline void dump_text_pointer(const T *pointer, const ApiDumpSettings &settings, const char *type_string, const char *name,
-                              int indents, std::ostream &(*dump)(const T, const ApiDumpSettings &, int, Args... args),
-                              Args... args) {
+template <typename T>
+void dump_text_pointer(const T *pointer, const ApiDumpSettings &settings, const char *type_string, const char *name, int indents,
+                       void (*dump)(const T, const ApiDumpSettings &, int)) {
     if (pointer == NULL) {
         settings.formatNameType(settings.stream(), indents, name, type_string);
         settings.stream() << "NULL\n";
     } else {
-        dump_text_value(*pointer, settings, type_string, name, indents, dump, args...);
+        dump_text_value(*pointer, settings, type_string, name, indents, dump);
     }
 }
 
-template <typename T, typename... Args>
-inline void dump_text_pointer(const T *pointer, const ApiDumpSettings &settings, const char *type_string, const char *name,
-                              int indents, std::ostream &(*dump)(const T &, const ApiDumpSettings &, int, Args... args),
-                              Args... args) {
+template <typename T>
+void dump_text_pointer(const T *pointer, const ApiDumpSettings &settings, const char *type_string, const char *name, int indents,
+                       void (*dump)(const T &, const ApiDumpSettings &, int)) {
     if (pointer == NULL) {
         settings.formatNameType(settings.stream(), indents, name, type_string);
         settings.stream() << "NULL\n";
     } else {
-        dump_text_value(*pointer, settings, type_string, name, indents, dump, args...);
+        dump_text_value(*pointer, settings, type_string, name, indents, dump);
     }
 }
 
-template <typename T, typename... Args>
-inline void dump_text_value(const T object, const ApiDumpSettings &settings, const char *type_string, const char *name, int indents,
-                            std::ostream &(*dump)(const T, const ApiDumpSettings &, int, Args... args), Args... args) {
+template <typename T>
+void dump_text_value(const T object, const ApiDumpSettings &settings, const char *type_string, const char *name, int indents,
+                     void (*dump)(const T, const ApiDumpSettings &, int)) {
     settings.formatNameType(settings.stream(), indents, name, type_string);
-    dump(object, settings, indents, args...) << "\n";
+    dump(object, settings, indents);
+    settings.stream() << "\n";
```

Comment:
not really your code, but here and elsewhere, why `"\n"` instead of `std::endl`?

Reply:
std::endl does _two_ things, add a new line and flush the output. I don't want to flush output unless explicitly asked for by the user, which is a separate option. Using "\n" allows for better performance in the general case cause it causes iostreams to only flush when necessary, and not on every line.